src/tools/clippy/clippy_lints/src/unnested_or_patterns.rs

```diff
@@ -178,7 +178,12 @@ fn unnest_or_patterns(pat: &mut P<Pat>) -> bool {
                     continue;
                 };
                 this_level_changed = true;
-                alternatives.splice(idx..=idx, inner);
+ 
+                // Equivalent to `alternatives.splice(idx..=idx, inner);`.
+                alternatives.remove(idx);
+                for p in inner.into_iter().rev() {
+                    alternatives.insert(idx, p);
+                }
```

Comment:
This looks algorithmically worse since it'll move the tail on every insert. The splice impl moves the tail by the size hint, inserts the elements, then moves it back.

I have no idea how hot this code is of course. Does clippy have some benchmarks?

Reply:
Not that I know of. `ThinVec` doesn't have `splice`, alas.

Reply:
So I assume this case will be uncommon to hit, especially uncommon with large enough N that it matters. But I still am concerned about letting an N² footgun lie around. At least a FIXME, an issue to get splice into thinvec or something would be good.

Or if thin-vec had cheap conversions between itself and vec then maybe that could be used.

With that addressed r=me.

Reply:
I filed https://github.com/Gankra/thin-vec/issues/39 about adding `ThinVec::splice` and added a FIXME comment that refers to it.

Cheap conversions between `ThinVec` and `Vec` aren't possible because the memory layout is different, alas.

Reply:
Good news: `thin-vec` 0.2.10 was just released and it includes support for `splice`. Thanks, @Gankra! So I have changed this back to the original code.